nodebuilder/header/opts.go

```diff
@@ -0,0 +1,26 @@
+package header
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/core"
+	header "github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
+	p2p "github.com/celestiaorg/celestia-node/libs/header/p2p"
+)
+
+// WithMetrics provides sets `MetricsEnabled` to true on ClientParameters for the header exchange
+func WithMetrics(ex libhead.Exchange[*header.ExtendedHeader]) error {
+	switch any(ex).(type) {
+	case *p2p.Exchange[*header.ExtendedHeader]:
+		exchange, ok := (ex).(*p2p.Exchange[*header.ExtendedHeader])
+		if !ok {
+			return fmt.Errorf("header.WithMetrics: type is to *p2p.Exchange but cast failed")
+		}
+		return exchange.RegisterMetrics()
+	case *core.Exchange:
+		return nil
+	default:
+		return fmt.Errorf("header.WithMetrics: unknown/unsupported (provided) exchange type")
+	}
```

Comment:
I think we agreed to avoid hard WithMetrics to have context of what other Exchanges exist and instead just give em check for one he knows about and do nothing otherwise.

Reply:
Aaah